crates/shared/src/recent_block_cache.rs

```diff
@@ -249,15 +289,21 @@ where
         }
 
         let cache_miss_block = block.unwrap_or(last_update_block);
-        let uncached_values = self
-            .fetch_inner(cache_misses.clone(), Block::Number(cache_miss_block))
-            .await?;
+        let cache_misses: Vec<_> = cache_misses.into_iter().collect();
+        // Splits fetches into chunks because we can get over 1400 requests when the
+        // cache is empty which tend to time out if we don't chunk them.
+        for chunk in cache_misses.chunks(200) {
+            let keys = chunk.iter().cloned().collect();
+            let (fetched, found_keys) = self
+                .fetch_inner_many(keys, Block::Number(cache_miss_block))
+                .await?;
+            cache_hits.extend_from_slice(&fetched);
 
-        cache_hits.extend_from_slice(&uncached_values);
-
-        {
             let mut mutexed = self.mutexed.lock().unwrap();
-            mutexed.insert(cache_miss_block, cache_misses.into_iter(), uncached_values);
+            mutexed.insert(cache_miss_block, chunk.iter().cloned(), fetched);
+            for key in found_keys {
+                mutexed.recently_used.cache_set(key, ());
```

Comment:
Instead of setting the recently_used cache in two places (`get` and `fetch`) can we just iterate over all results at the end and set recently used if we have an Ok result?

Reply:
This would not work in cases where we have to fetch so much liquidity that we can't do it in the 15s solve deadline. In that case we would simply time out and never consider the maybe 80% of the keys we were able to fetch as requested. This would then mean the background task will not fetch them and that could result in us not being able to fetch all the liquidity for a long time if we are unlucky.